tests/build/build.go

```diff
@@ -991,15 +993,24 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
 					return true, nil
 				}, time.Minute*2, constants.PipelineRunPollingInterval).Should(BeTrue(), "timeout while checking if any more pipelinerun is triggered")
 
-				// Check build status annotation
-				component, err = f.AsKubeAdmin.HasController.GetComponent(componentName, testNamespace)
-				Expect(component).ToNot(BeNil())
-				Expect(err).ShouldNot(HaveOccurred())
-
-				statusBytes := []byte(component.Annotations[controllers.BuildStatusAnnotationName])
 				buildStatus := &controllers.BuildStatus{}
-				Expect(json.Unmarshal(statusBytes, buildStatus)).To(Succeed())
-				Expect(buildStatus.Message).To(Equal(fmt.Sprintf("unexpected build request: %s", invalidAnnotation)))
+				Eventually(func() error {
+					component, err = f.AsKubeAdmin.HasController.GetComponent(componentName, testNamespace)
+					if err != nil {
+						return err
+					} else if component == nil {
+						return fmt.Errorf("got component as nil after getting component %s in namespace %s", componentName, testNamespace)
+					}
+					statusBytes := []byte(component.Annotations[controllers.BuildStatusAnnotationName])
```

Comment:
If the annotation doesn't exist, it would cause json unmarshalling error.

Reply:
agree on this.. need to be handled gracefully, Would like to address this with another PR, including all the other places we are Unmarshalling in the build tests